File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/dto/ModelProviderCreateRequest.java
package com.vipamp.vipclaw.admin.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * 模型服务商创建请求 DTO
 *
 * @author vipamp
 * @since 2026-03-13
 */
@Data
@Schema(description = "模型服务商创建请求对象")
public class ModelProviderCreateRequest {

    /**
     * 服务商名称（dashscope/openai/ollama）
     */
    @Schema(description = "服务商名称（dashscope/openai/ollama）", example = "dashscope", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "服务商名称不能为空")
    @Size(max = 50, message = "服务商名称长度不能超过 50 个字符")
    private String name;

    /**
     * 显示名称
     */
    @Schema(description = "显示名称", example = "阿里云 DashScope", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "显示名称不能为空")
    @Size(max = 100, message = "显示名称长度不能超过 100 个字符")
    private String displayName;

    /**
     * API 密钥
     */
    @Schema(description = "API 密钥", example = "sk-xxxxxxxxxxxxxxxx")
    private String apiKey;

    /**
     * API 地址
     */
    @Schema(description = "API 地址", example = "https://dashscope.aliyuncs.com/compatible-mode/v1")
    private String baseUrl;

    /**
     * 状态（0:禁用，1:启用）
     */
    @Schema(description = "状态（0:禁用，1:启用）", example = "1")
    private Integer status;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/dto/ModelProviderCreateRequest.kt
package com.vipamp.vipclaw.admin.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.*

/**
 * Model provider creation request object
 */
@Schema(description = "Model provider creation request object")
data class ModelProviderCreateRequest(
    @field:NotBlank(message = "Provider type cannot be empty")
    @field:Size(min = 1, max = 50, message = "Provider type length must be between 1-50 characters")
    @field:Pattern(regexp = "^[a-z0-9_]+$", message = "Provider type can only contain lowercase letters, numbers and underscores")
    @Schema(description = "Provider technical type", example = "dashscope", requiredMode = Schema.RequiredMode.REQUIRED)
    val type: String,

    @field:NotBlank(message = "Name cannot be empty")
    @field:Size(min = 1, max = 100, message = "Name length must be between 1-100 characters")
    @Schema(description = "Provider name", example = "Alibaba Cloud Bailian", requiredMode = Schema.RequiredMode.REQUIRED)
    val name: String,

    @field:Size(max = 500, message = "Description length cannot exceed 500 characters")
    @Schema(description = "Provider description", example = "Large language model API service provided by Alibaba Cloud")
    val description: String? = null,

    @field:Size(max = 500, message = "API key length cannot exceed 500 characters")
    @Schema(description = "API key (sensitive information)", example = "sk-xxxxxxxxxxxxxxxx")
    val apiKey: String? = null,

    @field:Pattern(
        regexp = "^(https?:\\/\\/)?([\\w.-]+)(:\\d+)?(\\/[^\\s]*)?$|^$",
        message = "Invalid API URL format",
    )
    @Schema(description = "API base URL (can be empty)", example = "https://dashscope.aliyuncs.com/compatible-mode/v1")
    val baseUrl: String? = null,

    @Schema(description = "Whether public (0:private 1:public)", example = "1")
    val isPublic: Int? = 1,
)
